zeebe/engine/src/main/java/io/camunda/zeebe/engine/processing/processinstance/ProcessInstanceMigrationMigrateProcessor.java

```diff
@@ -320,6 +320,8 @@ private void tryMigrateElementInstance(
 
     final List<DirectBuffer> activeSequenceFlowIds = elementInstance.getActiveSequenceFlowIds();
     if (!activeSequenceFlowIds.isEmpty()) {
+      checkSequenceFlowIdsIntegrity(elementInstance, processInstanceKey);
+
```

Comment:
❌ I had another thought and concluded that we cannot use the `activeSequenceFlowIds` to determine which sequence flow instances should be migrated, because by definition it does not support process instances created before 8.6.

💡 Instead, we can use your newly created method `ElementInstanceState.visitTakenSequenceFlows` to visit them. This provides all the needed info reliably as it is build around the `numberOfTakenSequenceFlowsColumnFamily` which has been the state tracking this for the joining parallel gateway since "forever". This would also mean we don't need a safety check

Reply:
@korthout Oh, I missed that as well. Alright, we had to determine active sequence flows using `numberOfTakenSequenceFlowsColumnFamily` data. I updated the code accordingly.

Note: `numberOfTakenSequenceFlowsColumnFamily` is only used for parallel and inclusive gateway. However, in precondition for concurrent command we need to make sure there are no active sequence flows to elements except parallel and inclusive gateway. Therefore, the check should stay the same in the concurrency precondition.